### PR TITLE
Feature/processing graph

### PR DESCRIPTION
**Pull request by lukasbutzmann**
Add operator for referencedata in processing graph

### CODE REVIEW

File: src/main/java/org/n52/wacodis/javaps/preprocessing/graph/TrainDataOperator.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+import org.n52.wacodis.javaps.WacodisProcessingException;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.feature.type.AttributeDescriptor;
+import org.opengis.feature.type.PropertyDescriptor;
+
+/**
+ *
+ * @author LukasButzmann
+ */
```
Comment (SebaDro) [inline src/main/java/org/n52/wacodis/javaps/preprocessing/graph/TrainDataOperator.java]: Please add javadoc

File: src/main/java/org/n52/wacodis/javaps/preprocessing/graph/TrainDataOperator.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                    iterator.close();
+                }
+
+                SimpleFeatureCollection simpleCollection = DataUtilities.collection(features);
+                return simpleCollection;
+            }else{
+                return(input);
+            }
+        }else{
+            String message = "Error! The Features in InputCollection don't have the Attribute <"+attributeName+">!";
```
Comment (SebaDro) [inline src/main/java/org/n52/wacodis/javaps/preprocessing/graph/TrainDataOperator.java]: No need to include "Error!" within the exception message as the exception means an error for itself. In addition, the message can be passed directly to the exception constructor without storing it within an extra variable.

**Review by SebaDro — CHANGES_REQUESTED** [review]
Please, apply the changes that are mentioned in the comments and add an extra test within `TrainDataOperatorTest` to test if an exception will be thrown if the class name is not available as attribute within the FeatureCollection.

File: src/main/java/org/n52/wacodis/javaps/preprocessing/graph/TrainDataOperator.java (hunk 1 above)
Comment (lukasbutzmann) [reply src/main/java/org/n52/wacodis/javaps/preprocessing/graph/TrainDataOperator.java]: add javadoc

**Pull request merged**